packages/expo-router/src/fork/useLinking.ts

```diff
@@ -156,7 +156,7 @@ export function useLinking(
 
       // Make sure that the routes in the state exist in the root navigator
       // Otherwise there's an error in the linking configuration
-      return state?.routes.some((r) => !rootState?.routeNames.includes(r.name));
+      return state?.routes.some((r) => !rootState?.routeNames?.includes?.(r.name));
```

Comment:
I had to add that change, otherwise I'd run into `Cannot read properties of undefined (reading 'includes')`
<img width="1372" alt="Screenshot 2025-06-23 at 23 13 51" src="https://github.com/user-attachments/assets/ce741141-099d-4adb-bc02-9627e523079f" />

It seems to be related to the createMemoryHistory. It seems like back button support is only working very sporadically with  the RouterStack, appreciate if someone could look into this.

Reply:
I'll look into it. There was an issue related to this problem I need to solve